src/xercesc/internal/ReaderMgr.cpp

```diff
@@ -72,8 +72,9 @@ ReaderMgr::~ReaderMgr()
     //  entities it still references!)
     //
     delete fCurReader;
+    if (fOwnEntity)
+        delete fCurEntity;
```

Comment:
`delete` checks for `nullptr`
```suggestion
    delete fCurEntity;
```